### DISCUSSION TITLE
[Question] package.json scripts字段脚本变量在src/browser/app.tsx文件中获取不到值

### DISCUSSION THREAD

**Discussion by 767746649zyy** [opensumi/core#2191] [Q&A]
ide-electron项目package.json scripts字段自定义脚本命令加入变量WORKSPACE=E://etest，在src/browser/app.tsx文件中获取不到自定义WORKSPACE变量值, pack命令构建后执行exe，获取不到变量值WORKSPACE，start命令启动获取的到变量值WORKSPACE
![image](https://user-images.githubusercontent.com/23164065/211480907-41b2fb42-caa8-44d3-a524-244f8a9e4915.png)
![image](https://user-images.githubusercontent.com/23164065/211481023-a37404e1-87d9-4a85-ab71-55ce320fbbbc.png)

**Comment by yantze** [comment]
electronEnv.env.WORKSPACE 这些变量是来自框架内部的值，不是通过 process.env.WORKSPACE 获取的，如果你需要用可以直接使用 process.env.WORKSPACE 

赋值代码 WORKSPACE_DIR 
https://github.com/opensumi/core/blob/0469c5f4d54ed00d8ba6d5bae352b1239bed9f55/packages/core-electron-main/browser-preload/index.js#L98

**Comment by 767746649zyy** [comment]
process.env.WORKSPACE的值 构建后得产物获取不到，但是在start启动阶段可以获取到此值，我想在构建命令里也加入WORKSPACE，并在构建后得产物里使用，是否可行？

**Comment by yantze** [comment]
> process.env.WORKSPACE的值 构建后得产物获取不到，但是在start启动阶段可以获取到此值，我想在构建命令里也加入WORKSPACE，并在构建后得产物里使用，是否可行？

「process.env.WORKSPACE的值 构建后得产物获取不到」 要通过命令行去启动，启动前加上环境变量就行了，比如这样
```shell
ABC=testVal /Applications/O2.app/Contents/MacOS/O2
```

**Comment by yantze** [comment]
如果是 Windows 环境，在应用中需要获取变量，可以下面这样：
<img width="571" alt="image" src="https://user-images.githubusercontent.com/2226423/211707936-71034131-995d-4eb1-9585-cdeb33016a46.png">

**Comment by 767746649zyy** [comment]
好的 我用的child_process.spawn设置的env
